fix: SO to PO flow improvement

marination opened the pull request:
## Enhancements on checkbox to create PO from SO
- Added checkbox for user to decide whether PO to be created for default supplier or a usual PO is to be created where user chooses supplier
  <img src="https://user-images.githubusercontent.com/25857446/94788844-ce6a3280-03f1-11eb-9298-a4826e7216d9.png" width="70%">

-  On enabling checkbox,only rows with Default Supplier against them will be visible.
   <img src="https://user-images.githubusercontent.com/25857446/94788832-cc07d880-03f1-11eb-9ccc-b9084ca929a7.png" width="70%">
- Supplier field is added which will show Default Suppliers against Items
- Pending Qty field added which will show pending qty to order. It will consider already ordered qty

## Behaviour

<details><summary>Checkbox disabled, normal PO creation</summary>

All items will be mapped to a new PO and user will be redirected there.
  ![so-to-po](https://user-images.githubusercontent.com/25857446/93582252-d4a6ea80-f9bf-11ea-89a4-b76d9a40c85d.gif)

</details>

<details><summary>Checkbox enabled</summary>

POs created for each Supplier. So, if there were two item rows belonging to the same supplier, both will come under one PO for their supplier

![def-supp-po](https://user-images.githubusercontent.com/25857446/94789300-5f410e00-03f2-11eb-8e3b-55a67ade84b4.gif)

</details>

<details><summary>Partial PO creation</summary>

The Pending qty column will show exact amount pending, pending amount will be mapped

  ![partial-so-to-po-supplier](https://user-images.githubusercontent.com/25857446/93582678-70d0f180-f9c0-11ea-85f5-e56f36bb50b6.gif)

</details>

marination committed:
fix: SO to PO flow improvement

marination committed:
fix: Dont map shipping_address

- shipping_address is a text field in SO and link field in PO
- Drop shipping case handles its mapping
- normal case doesnt need to map

Anurag810 reviewed (commented):
please also take this use case under consideration.
https://github.com/frappe/erpnext/pull/23341

marination committed:
fix: Hide/Add rows depending on Against Default Supplier

marination committed:
fix: Removed Default Supplier Select field from popup

- removed Default Supplier Select field from popup
- only loop through suppliers of selected items if via default supplier
- only check for items in selected items

marination committed:
Merge branch 'develop' into po-to-so-ux

marination committed:
Merge branch 'develop' into po-to-so-ux

nabinhait committed:
Merge branch 'develop' into po-to-so-ux

marination committed:
fix: Sales Order Drop Shipping Test

marination committed:
fix: (translation)Multi line to single line strings

marination committed:
Merge branch 'develop' into po-to-so-ux

nabinhait committed:
Merge branch 'develop' into po-to-so-ux

marination commented on erpnext/selling/doctype/sales_order/sales_order.py:
```diff
@@ -822,24 +816,21 @@ def set_missing_values(source, target):
+	suppliers = [item.get('supplier') for item in selected_items if item.get('supplier') and item.get('supplier')]
+	suppliers = list(set(suppliers))
+
+	items_to_map = [item.get('item_code') for item in selected_items if item.get('item_code') and item.get('item_code')]
```
🫤
